File: 9-palindrome-number/palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        int temp=x;
        int rev=0;
        while(x>0){
            int n=x%10;
            rev=rev*10+n;
             x=x/10;
        }
        if(x<0){
            return false;
        }
        if(temp==rev){
            return true;
        }
        return false;
        
    }
}

File: 9-palindrome-number/palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        bool ans=false;
        if(x<0){return ans;}
        long  temp=x;
       long  rev=0;
        while(x>0){
            int dig=x%10;
            rev=(rev*10)+dig;
            x=x/10;
        }
        if(temp==rev){ans=true;}
        return ans;
    }
};